app/Main.hs

```diff
@@ -125,51 +115,62 @@ data Message
 
 main :: IO ()
 main = do
-  Opts {..} <- execParser parserInfo
+  Opts{..} <- execParser parserInfo
   C.Config{..} <- Yaml.loadYamlSettings [optsConfigFile] [] Yaml.useEnv
   chan <- newTChanIO
   let task rf = Task { taskDeployPath    = configDeployPath
                      , taskSource        = configSource
                      , taskReleaseFormat = rf }
   let printFnc dest str = atomically $
         writeTChan chan (PrintMsg dest str)
-      hap shell sshOpts =  do
+      hap shell sshOpts = do
         r <- Hap.runHapistrano sshOpts shell printFnc $
           case optsCommand of
-            Deploy cliReleaseFormat cliKeepReleases -> do
+            Deploy cliReleaseFormat cliKeepReleases cliKeepOneFailed ->
```

Comment:
I think keeping the previous `-> do` here and removing the `in do` below should make the previous indentation work here, which might make it easier to review.

Reply:
The reason I'm doing this is because I want the `failStateAndThrow` exception handler to be in the outer scope to catch the exceptions that happen within the `do` you mention; otherwise it would remain in the inner scope and can't be used outside.

Reply:
Can it be moved to around line 126 then (same level as `hap` and `printFnc`)?

Reply:
I don't think so, because this can only work if we have been asked to run the `deploy` command, that's why it's within the case-splitting.